designs/auto-certs-renewal.md

```diff
@@ -0,0 +1,122 @@
+# Auto Certificates Renewal
+
+## Context
+
+As a user who deployed K8s cluster with EKS-A, I want an option to automatically renew the certificates in etcd and control plane nodes before expiry in 1 year.
+
+### External ETCD node certificates
+
+* etcd node’s certificates are stored in /etc/etcd/pki (/var/lib/etcd/pki in bottlerocket)
+* each certificate comes in pair: key and crt
+* all etcd nodes contain the same key pair names no matter if it’s the first node (not in term of contents)
+
+list of all the certificate files:
+
+1. apiserver-etcd-client pair: generated by etcdadm,  apiserver can use it to access etcd, **but it’s not used**
+2. ca pair: the only pair which is not generated by etcdadm, it comes from cloud-init script.
+3. etcdctl-etcd-client pair: generated by etcdadm during `etcdadm join <endpoint>`
+4. peer pair: generated by etcdadm `etcdadm join <endpoint>`
+5. server pair: generated by etcdadm `etcdadm join <endpoint>`
+
+### CP node certificates
+
+* all certificates are located in /etc/kubernetes/pki  (/var/lib/kubeadm/pki in bottlerocket)
+* all cp nodes contains the same key pair names
+
+list of all the certificate files:
+
+1. apiserver-etcd-client pair: from cloud-init script, but the source is etcdadm controller 
+2. ca pair: from cloud-init script
+3. front-proxy-ca pair: from cloud-init script
+4. sa pair: from cloud-init script
+5. etcd/ca.crt: from cloud-init script
+6. apiserver-kubelet-client pair: from kubeadm
+7. apiserver pair: from kubeadm
+8. front-proxy-client pair: from kubeadm
+
+The above certificate files are described in the official doc: https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
+
+### Worker node certificates
+
+1. just ca.crt in /etc/kubernetes/pki, which should be generated after `kubeadm join`
+
+## Goal
+
+1. Auto renew all the above mentioned certificates except etcd CA and kube CA, which last for 10 years.
```

Comment:
Do we have an open issue for handling this also? Or, perhaps there's an easy win to address this at the same time?

Reply:
No such issue was created as far as I know.

Reply:
Are we able to handle it at the same time. I know we want to address the first problem, but both problems seem similar.

Reply:
is 10 years configurable? what happens if there is a need to roll the certs before then? what if they dont want 10 years.